crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -202,18 +250,54 @@ pub struct Rectangle {
 impl Primitive2d for Rectangle {}
 
 impl Rectangle {
-    /// Create a rectangle from a full width and height
+    /// Create a new `Rectangle` from a full width and height
     pub fn new(width: f32, height: f32) -> Self {
-        Self::from_size(Vec2::new(width, height))
+        Self {
+            half_width: width / 2.0,
+            half_height: height / 2.0,
+        }
     }
 
-    /// Create a rectangle from a given full size
+    /// Create a new `Rectangle` from a given full size
     pub fn from_size(size: Vec2) -> Self {
+        Self::from_half_size(size / 2.0)
+    }
+
+    /// Create a new `Rectangle` from a given half-size
+    pub fn from_half_size(half_size: Vec2) -> Self {
         Self {
-            half_width: size.x / 2.,
-            half_height: size.y / 2.,
+            half_width: half_size.x,
+            half_height: half_size.y,
         }
     }
```

Comment:
`Cuboid` stores a `half_size` vector, but `Rectangle` stores it with separate properties. Should `Rectangle` store a `half_size` as well? It would make this constructor obsolete and make the shape representations more consistent.